pkg/reconciler/configuration/configuration.go

```diff
@@ -237,21 +254,23 @@ func CheckNameAvailability(config *v1alpha1.Configuration, lister listers.Revisi
 	return rev, nil
 }
 
-func (c *Reconciler) latestCreatedRevision(config *v1alpha1.Configuration) (*v1alpha1.Revision, error) {
-	if rev, err := CheckNameAvailability(config, c.revisionLister); rev != nil || err != nil {
-		return rev, err
-	}
+func (c *Reconciler) getSortedCreatedRevisions(config *v1alpha1.Configuration) ([]*v1alpha1.Revision, error) {
+	_, nameAvailabilityErr := CheckNameAvailability(config, c.revisionLister)
```

Comment:
I see. If I don't make this change here though, it would return immediately when it gets a match, so if that matching revision is not ready, we would not update the latest ready revision to the last ready revision.